langchain4j-agentic/src/test/java/dev/langchain4j/agentic/JsonInMemoryCognispherePersistenceProvider.java

```diff
@@ -0,0 +1,51 @@
+package dev.langchain4j.agentic;
+
+import dev.langchain4j.agentic.cognisphere.Cognisphere;
+import dev.langchain4j.agentic.cognisphere.CognisphereKey;
+import dev.langchain4j.agentic.cognisphere.CognispherePersistenceProvider;
+import dev.langchain4j.agentic.cognisphere.CognisphereSerializer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class JsonInMemoryCognispherePersistenceProvider implements CognispherePersistenceProvider {
+
+    private final Map<CognisphereKey, String> jsonCognispheres = new HashMap<>();
+    private final List<Object> loadedIds = new ArrayList<>();
+
+    @Override
+    public boolean save(Cognisphere cognisphere) {
+        jsonCognispheres.put(cognisphere.key(), CognisphereSerializer.toJson(cognisphere));
```

Comment:
What is actually the point of serializing to JSON if it's just stored in memory? I mean, when retrieving it using `load`, it will be deserialized again. This JSON step looks like unnecessary overhead. 
If you want to prevent the Cognisphere from being changed by something external, you could make it `Cloneable` and create a clone of it before storing.

Reply:
This class is only for testing purposes, to demonstrate and validate how the cognisphere registry interact with the persistence layer. I also want the Cognisphere to be serializable in json and this allowed me to even test this.

Reply:
Oh, dumb me, I missed that this is for a test :)